ldk/javascript/src/types/oliveHelps/index.d.ts

```diff
@@ -201,6 +222,55 @@ declare namespace OliveHelps {
     headers: Record<string, string[]>;
   }
 
+  //--Search
+  interface Search {
+    createIndex: ReadableWithThreeParams<string, Array<Document>, Config, Index>;
+    openIndex: ReadableWithTwoParams<string, Config, Index>;
+    exists: ReadableWithParam<string, boolean>;
+  }
+
+  interface Index {
+    search(
+      term: string,
+      callback: (error: Error | undefined, searchResult: SearchResult) => void,
+    ): void;
+    queryStringSearch(
+      queryString: string,
+      callback: (error: Error | undefined, searchResult: SearchResult) => void,
+    ): void;
+    update(
+      documents: Array<Document>,
+      config: Config,
+      callback: (error: Error | undefined) => void,
+    ): void;
+    delete(callback: (error: Error | undefined) => void): void;
+  }
```

Comment:
I think this should work for you as well: 
`interface Index {
    search: ReadableWithParam<string, SearchResult>;
    queryStringSearch: ReadableWithParam<string, SearchResult>;
    update: ReadableWithTwoParams<Array<Document>, Config, void>;
    delete: Readable<void>;
  }`

Reply:
should be completed